src/query/users/src/user_mgr.rs

```diff
@@ -43,24 +43,17 @@ impl UserApiProvider {
                     UserInfo::new_no_auth(&user.username, &user.hostname)
                 }
             };
-            if user.is_localhost() {
-                user_info.grants.grant_privileges(
-                    &GrantObject::Global,
-                    UserPrivilegeSet::available_privileges_on_global(),
-                );
-                user_info
-                    .grants
-                    .grant_role(BUILTIN_ROLE_ACCOUNT_ADMIN.to_string());
-                user_info
-                    .option
-                    .set_default_role(Some(BUILTIN_ROLE_ACCOUNT_ADMIN.to_string()));
-                user_info.option.set_all_flag();
-            } else {
-                return Err(ErrorCode::UnknownUser(format!(
-                    "only accept root from localhost, current: '{}'@'{}'",
-                    user.username, user.hostname
-                )));
-            }
+            user_info.grants.grant_privileges(
```

Comment:
I guess checking the root user's original IP to be localhost is still a must have, currently we restricted the root access by localhost only. If this behaviour changed, it might leaks some security risks.